wicketd/src/update_tracker.rs

```diff
@@ -1134,14 +1089,61 @@ impl UpdateContext {
         &self,
         cx: &StepContext,
         mut ipr_receiver: mpsc::Receiver<EventReport<InstallinatorSpec>>,
-    ) -> anyhow::Result<()> {
+    ) -> anyhow::Result<WriteOutput> {
+        let mut write_output = None;
+
         while let Some(report) = ipr_receiver.recv().await {
+            // Prior to processing the report, check for the completion metadata
+            // that indicates which disks installinator attempt to /
+            // successfully wrote. We only need to do this if we haven't already
+            // seen the metadata we care about in a previous report; we should
+            // never get multiple completion events that differ in this
+            // metadata.
+            if write_output.is_none() {
+                for event in &report.step_events {
+                    // We only care about the outcome of completion events.
+                    let outcome = match &event.kind {
+                        StepEventKind::StepCompleted { outcome, .. }
+                        | StepEventKind::ExecutionCompleted {
+                            last_outcome: outcome,
+                            ..
+                        } => outcome,
+                        StepEventKind::NoStepsDefined
+                        | StepEventKind::ExecutionStarted { .. }
+                        | StepEventKind::ProgressReset { .. }
+                        | StepEventKind::AttemptRetry { .. }
+                        | StepEventKind::ExecutionFailed { .. }
+                        | StepEventKind::Nested { .. }
+                        | StepEventKind::Unknown => continue,
+                    };
```

Comment:
Wondering how you would feel about making this a method on `StepEventKind`.

Reply:
Yeah, seems perfectly reasonable. Moved in 7c6dceef